UXD-1054 Refactor Textarea API, styles + docs 🐐

mikrotron opened the pull request:
### Purpose 🚀
Remove `classNames` and refactor styling to pure `styled-components` conventions.
Conform API to be consistent with `<Input>` and `<Select>`.


### Changeset 📋
#### Changed

- Refactored styling to remove classes from markup and utilize styled components.
- Allow a number or string value for `maxHeight` with the default value changed to `300` instead of `300px`.

#### Removed

- Removed `className` prop that added a class to root `<div>` element.

#### Added

- Added `<Textarea.Container>` prop collector to allow classes and other attributes to be added to root `<div>` element.
- Added `minHeight` prop to set a height less than the hard-coded `80px` value.

#### Migration Guide

- If a custom class name is required on the root element, it can be added like `<Textarea><Textarea.Container className="custom"></Textarea>`.
- If `classNames` were used to target DOM elements or identify states of the `<Textarea>`, they will need to be updated to use attributes like `data-pka-anchor`, `aria-invalid`, etc.


### Notes ✏️
- Also updated Storybook documentation – updated to CSF and improved Screener coverage.
- Also harmonized some README and Storybook documentation between `<Input>`, `<Select>` and `<Textarea>`.


### Storybook 📕
http://storybooks.highbond-s3.com/paprika/UXD-1054--refactor-textarea/?path=/story/forms-textarea-docs--page


### Screenshots 📸
New **Variations** story
<img width="276" alt="Screen Shot 2021-05-20 at 8 34 14 PM" src="https://user-images.githubusercontent.com/14944896/119086795-e1e6ff80-b9ba-11eb-9b81-8773083abaf7.png">
<img width="322" alt="Screen Shot 2021-05-20 at 8 35 04 PM" src="https://user-images.githubusercontent.com/14944896/119086801-e3b0c300-b9ba-11eb-8fe1-a373bb1620f8.png">


### References 🔗
https://aclgrc.atlassian.net/browse/UXD-1054

Companion `<Input>` PR: https://github.com/acl-services/paprika/pull/1003

Companion `<Select>` PR: https://github.com/acl-services/paprika/pull/1045




<!--
### Resources 🔖

Paprika README —
https://github.com/acl-services/paprika/blob/master/README.md

Contributing Guidelines —
https://github.com/acl-services/paprika/wiki/Contributing-Guidelines

Conventional Commits —
https://www.conventionalcommits.org/

Ask for help —
https://github.com/acl-services/paprika/issues/new?template=help_wanted.md

-->

nahumzs commented on packages/Textarea/README.md:
```diff
@@ -22,56 +22,51 @@ npm install @paprika/textarea
-  />
-);
+<Textarea
+  defaultValue="Hello world"
```
To comply with the example of uncontrrolled components in react documentation I think wee should do something like this:


```js
const refTextarea = React.useRef(null)

<Textarea ref={refTextarea} defaultValue="hi!" />

refTextarea.current // hi!

```

using the onChange will imply that you have to store dom value in some state to do something with it or to submit it.

mikrotron replied:
Good point.  I think that makes sense, but I'll want to update the Uncontrolled story as well and also update the `<Select>` and `<Input>` to make them all consistent.

allison-c commented:
>Selectors may have to change since there is no longer a <div> element wrapping the <textarea>.

I feel we better make this consistent in the future. For a `Select` component, we changed it to have a `div` wrapper. For `TextArea` we removed the `div` wrapper. It's not easy for consumers to remember which components have which don't. Just always have the wrapper might be easier to understand.

allison-c commented on packages/Textarea/src/Textarea.js:
```diff
@@ -1,138 +1,139 @@
-      window.removeEventListener("resize", resize);
+      window.removeEventListener("resize", handleResize);
     };
   }, []);
```
missing `canExpand ` in the dependency list

allison-c replied:
and `handleResize ` too

mikrotron commented:
> > Selectors may have to change since there is no longer a  element wrapping the <textarea>.
> 
> I feel we better make this consistent in the future. For a `Select` component, we changed it to have a `div` wrapper. For `TextArea` we removed the `div` wrapper. It's not easy for consumers to remember which components have which don't. Just always have the wrapper might be easier to understand.

I debated that change too – consistency vs simplicity.  It's a close call, but I see @nahumzs has 👍 your suggestion so there seems to be more support for consistency. I will update it.

allison-c commented:
> > > Selectors may have to change since there is no longer a  element wrapping the <textarea>.
> > 
> > 
> > I feel we better make this consistent in the future. For a `Select` component, we changed it to have a `div` wrapper. For `TextArea` we removed the `div` wrapper. It's not easy for consumers to remember which components have which don't. Just always have the wrapper might be easier to understand.
> 
> I debated that change too – consistency vs simplicity. It's a close call, but I see @nahumzs has 👍 your suggestion so there seems to be more support for consistency. I will update it.

No need to update in this PR 😄

mikrotron commented:
> > > > Selectors may have to change since there is no longer a  element wrapping the <textarea>.
> > > 
> > > 
> > > I feel we better make this consistent in the future. For a `Select` component, we changed it to have a `div` wrapper. For `TextArea` we removed the `div` wrapper. It's not easy for consumers to remember which components have which don't. Just always have the wrapper might be easier to understand.
> > 
> > 
> > I debated that change too – consistency vs simplicity. It's a close call, but I see @nahumzs has 👍 your suggestion so there seems to be more support for consistency. I will update it. 
<br />

> No need to update in this PR 😄

I wouldn't want to introduce another breaking change later, so I've updated this PR.
